Adds guidelines for writing docs (#5305)
gravitational/teleport: docs/5.0/docs/best-practices.md

# Docs Best Practices

## Content

**Keep it simple**

Read the book [On Writing Well](https://www.amazon.com/Writing-Well-Classic-Guide-Nonfiction/dp/0060891548).

**Write for the target audience**

> Do not overload quickstart guides with content and complex terminology, focus

Reviewer: ```suggestion
Do not overload "getting started" guides with content and complex terminology, focus
```

😁

> on benefits to the user. Do not add architecture diagrams in quickstarts.

Reviewer: ```suggestion
on benefits to the user. Do not add architecture diagrams in "getting started" guides.
```

Likewise, do not do demos in architecture articles.

**Keep guides self-contained**

> All guides should have all what is needed for a reader

Reviewer: ```suggestion
All guides should have everything that is needed for a reader
```

> Quickstarts and guides articles should have all what is needed for a reader

Reviewer: ```suggestion
Getting started guides should have all what is needed for a reader
```

to accomplish their goal. This sometimes leads to duplication.
Use include directives for repeated content. Remove backslash before the exclamation marks for it to work:

```bash
{% raw %}{{\!CHANGELOG.md\!}}{% endraw %}
```

## Diagrams

Use [Teleport's lucidchart library](https://app.lucidchart.com/lucidchart/dfcf1f4a-5cf0-4758-8ebb-f6ea86900aba/edit)
to create diagrams with a consistent design language.

## Videos

> Mac users should use Quicktime's `Cmd-Shift-5` to record small part of the screen:

Reviewer: ```suggestion
Mac users should use Quicktime's `Cmd-Shift-5` to record a small part of the screen:
```

> Mac users should use quicktime's `Cmd-Shift-5` to record small part of the screen:

Reviewer: ```suggestion
Mac users should use Quicktime's `Cmd-Shift-5` to record small part of the screen:
```

![quicktime](../img/docs/quicktime.webp)

**Convert Mov to MP4 and WebM**

Quicktime outputs large `.mov` files. Use `ffmpeg` to convert them into `mp4` and `webm`
web-friendly formats:

```bash
# create MP4

> ffmpeg -i input.mov -b:v 0 -crf 25  output.mp4

Reviewer: ```suggestion
ffmpeg -i input.mov -b:v 0 -crf 25 output.mp4
```

> ffmpeg -i input.mov  -b:v 0 -crf 25  output.mp4

Reviewer: ```suggestion
ffmpeg -i input.mov -b:v 0 -crf 25 output.mp4
```

# create WebM
ffmpeg -i input.mov -c vp9 -b:v 0 -crf 41 output.webm
```

**Embed videos**

```html
<video autoplay loop muted playsinline>
  <source src="https://goteleport.com/teleport/videos/database-access-preview/dbaccessdemo.mp4" type="video/mp4" />
  <source src="https://goteleport.com/teleport/videos/database-access-preview/dbaccessdemo.webm" type="video/webm" />
Your browser does not support the video tag.
</video>
```

> To test videos locally add them to `img/video` folder of the docs.

Reviewer: ```suggestion
To test videos locally, add them to the `img/video` folder of the docs.
```

> For production upload to the website and refer from there:

Reviewer: ```suggestion
For production, upload to the website and refer from there:
```
